cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -426,7 +428,7 @@ renderTxIn (TxIn txId (TxIx ix)) =
   serialiseToRawBytesHexText txId <> "#" <> Text.pack (show ix)
 
 
-newtype TxIx = TxIx Word
+newtype TxIx = TxIx Word16
```

Comment:
This is also used in Byron transactions so if we we should do the appropriate check similarly to below: `guard (fromIntegral txix <= maxByronTxInIx) ?! TxBodyInIxOverflow` in `makeShelleyTransactionBody`.

Reply:
OK. That means that `toShelleyTxIn` can overflow, which feels bad, but I'll change for consistency.